### PR TITLE
pango: 1.47.0 → 1.48.3

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
Switches from gtk-doc to gi-doc for docs.

https://gitlab.gnome.org/GNOME/pango/-/blob/1.48.3/NEWS#L1-34

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/pango/default.nix [hunk 1]
```diff
@@ -1,57 +1,88 @@
...
+    libintl
+    harfbuzz
+  ] ++ lib.optionals x11Support [
+    libXft
+  ];
 
   mesonFlags = [
-    "-Dgtk_doc=${if stdenv.isDarwin then "false" else "true"}"
+    "-Dgtk_doc=true"
   ] ++ lib.optionals stdenv.isDarwin [
```
Comment (kubek2k) [inline pkgs/development/libraries/pango/default.nix:71]: this needs refinement to `lib.optionals !x11Support`

Comment (kubek2k) [reply pkgs/development/libraries/pango/default.nix:71]: with above change everthing works fine (need to override cairo and harfbuzz, to get rid of X libs)

Comment (jtojnar) [reply pkgs/development/libraries/pango/default.nix]: Done.

**Comment by jtojnar** [comment]
Had to play with doc output a bit but Devhelp now works.

image

**Comment by SuperSandro2000** [comment]
Sadly I can't test this because libxml is broken staging.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116347` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>pango</li>
</ul>
</details>

**Pull request merged**